### ISSUE TITLE
Have classes for all/core o-colors in the internal brand

### ISSUE THREAD

**Issue by wheresrhys** [Financial-Times/o-labels#102]
Time and time again I find myself defining o-labels modifier classes because the handful of internal use case classes defined in the internal brand don't cover nearly all the use cases we have. I also find myself hacking around the limitations by using the existing use case classes for other purposes.

While for external use I can see the benefit of restricting to predefined use cases, as brand consistency is very important. But internally, the multitude of different use cases makes me think trying to follow a use case based class strategy is not realistic, and a colour based one would meet the needs of users of the internal brand far better, while also making it more likely that colour combinations used would be accessible.

**Comment by wheresrhys** [comment]
I suppose this is a competing proposal to https://github.com/Financial-Times/o-labels/issues/58

**Comment by wheresrhys** [comment]
Perhaps a good middle ground is to define some general reusable classes based on a variety of scales which can be reapplied in lots of contexts, e.g.

### Price
- Luxury - purple
- Expensive - oxford
- Quality - Jade
- Budget - Mandarin

### 'Goodness'
- very bad - Red
- Bad - mandarin
- suboptimal - yellow
- neutral - grey
- good - wasabi
- very good - jade

Other possible scales include age, 'activeness' ... there were others too but I forget

**Comment by notlee** [comment]
Instead of a class for each colour combination we could look at using CSS Custom Properties for this, to keep bundle size low and options high. O&R don't support IE11 right @wheresrhys?

**Comment by JakeChampion** [comment]
>Origami team member:
>:wave: Does anyone know the browser support policy for O&R systems? I'm hoping Internet Explorer is not supported :crossed_fingers:
>
>Reliability team member:
>I'm not aware such a thing exists. I have not built anything that targets older browsers since joining the team!
>
> Reliability team member:
>Not clearly defined, but definitely doesn't include IE
>
> -- https://financialtimes.slack.com/archives/C07B3043U/p1608305407401400?thread_ts=1608301242.400900&cid=C07B3043U

They do not support IE11 🎉

**Comment by wheresrhys** [comment]
TBH, that's not really an advantage over us referencing the o-colors custom properties - we'd still need to create classes and come up with accessible text colours ourselves. Unless i"m missing something. Maybe you could provide an example of what you mean.

This would be internal brand only, where arguably bundle size is less of a concern, and is the css for each label all that bulky anyway?

**Comment by chee** [comment]
hey @wheresrhys :) do you have a link to some of the classes you've defined which we could add to o-labels?

i'm up for adding the goodness scale to the internal brand.

@notlee @JakeChampion 
we could also allow users to control the bundle size by defining the scales in a mixin options that's enabled by default for the internal brand, right?

i think the point Rhys makes here about needing to pick an accessible background/text color combo is an important one

**Comment by JakeChampion** [comment]
>  we could also allow users to control the bundle size by defining the scales in a mixin options that's enabled by default for the internal brand, right?

Yes, that's a good idea 😄

**Comment by wheresrhys** [comment]
Thanks for following this up.

Are any internal brand users expressing concern over the CSS bundle size? While your solution sounds neat, if it's a lot of effort I don't know if it's that big a concern.

Examples:

- https://github.com/Financial-Times/rel-eng-repos/blob/master/src/components/Matrix.jsx#L78-L123
- https://github.com/Financial-Times/biz-ops-admin/blob/master/src/browser/main.css#L77-L100
- https://github.com/Financial-Times/biz-ops-admin/blob/master/src/templates/components/primitives.jsx#L6-L28
- https://github.com/Financial-Times/risk-register/blob/main/public/styles.css#L4-L47

**Comment by notlee** [comment]
We chatted about this in our review of the project board. 

We decided:
- Release modifier classes for each label colour, with accessible text colour. They will have an option in the primary mixin and will be usable right away.
- Talk to the design team about colour usecases like price and goodness. Potentially add CSS aliases/demos/docs for these. Potentially define them in o-colors as usecases, so good/bad colours can be shared across components for example.
